### PR TITLE
Add `device` parameter to convert functions and update usage model de…

### PR DESCRIPTION
**Pull request by ayasyrev**
Add `device` parameter to convert functions and update usage model device in clip_similarity_scores.
related https://github.com/iterative/datachain-examples/issues/6

### CODE REVIEW

File: src/datachain/lib/clip.py [hunk 1]
```diff
@@ -130,17 +130,20 @@ def clip_similarity_scores(
         ```
     """
 
+    device = next(model.parameters()).device
```
Comment (dberenbaum) [inline src/datachain/lib/clip.py:133]: Probably worth noting for future reference that this is not guaranteed to work but is likely good enough for almost all use cases. See https://stackoverflow.com/questions/58926054/how-to-get-the-device-type-of-a-pytorch-module-conveniently.

Comment (dberenbaum) [reply src/datachain/lib/clip.py:133]: Maybe the function should also include a `device` parameter but default to this if no device is passed?

**Review by dberenbaum — COMMENTED** [review]
Looks great! See one comment about adding a `device` parameter to `clip_similarity_scores()`. Thanks for the contribution!

File: src/datachain/lib/clip.py (hunk 1 above)
Comment (ayasyrev) [reply src/datachain/lib/clip.py:133]: I suggest this first. Let's add `device`.

**Comment by dberenbaum** [comment]
One more thought: I think we should also add a `device` parameter to `DataChain.to_pytorch()` for model training. Looking at the clip fine tuning example, the image, text, and label need to be sent to the correct device during training.

@ayasyrev Let me know if you are up for this or would prefer to leave it out of this PR. If you want to take a look, most of the code is in https://github.com/iterative/datachain/blob/main/src/datachain/lib/pytorch.py.

**Comment by ayasyrev** [comment]
> One more thought: I think we should also add a `device` parameter to `DataChain.to_pytorch()` for model training. Looking at the clip fine tuning example, the image, text, and label need to be sent to the correct device during training.
> 
> @ayasyrev Let me know if you are up for this or would prefer to leave it out of this PR. If you want to take a look, most of the code is in https://github.com/iterative/datachain/blob/main/src/datachain/lib/pytorch.py.
Will check it.
Usually (OK, IMHO) move "batch" to `device`  used after dataloader. Will look for implementation.

**Comment by dberenbaum** [comment]
> Usually (OK, IMHO) move "batch" to `device` used after dataloader. Will look for implementation.

Yes, I guess this is needed in complex use cases where there are multiple devices. However, I think it makes sense to have a simple way to do this similar to https://huggingface.co/docs/datasets/en/use_with_pytorch.

**Comment by ayasyrev** [comment]
> Yes, I guess this is needed in complex use cases where there are multiple devices. However, I think it makes sense to have a simple way to do this similar to https://huggingface.co/docs/datasets/en/use_with_pytorch.
OK, check this example. 
Can do it.
Do you think to add it to this pr or better do another one?

**Comment by dberenbaum** [comment]
> Do you think to add it to this pr or better do another one?

Whatever you prefer @ayasyrev. If you want to do a separate PR, let me know.

**Review by dberenbaum — APPROVED** [review]

**Comment by dberenbaum** [comment]
Could you please fix the tests that got broken @ayasyrev? The tests use a mock for the model to speed up the tests, so you likely need to add the `parameters` attribute to the mocked model [here](https://github.com/iterative/datachain/blob/main/tests/unit/lib/conftest.py).
